### PR TITLE
Feature/py executor test

### LINKED ISSUE #4902: [Python] Test Executor
Add unit test of Python executor.

It will use Python Executor and create a `mul` operator and perform unit test of it.

### PR DESCRIPTION
**Pull request by reyoung**
Fix #4902

### CODE REVIEW

File: python/paddle/v2/framework/framework.py [hunk 1]
```diff
@@ -256,7 +256,8 @@ def __init__(self,
                     self.desc.set_block_attr(attr_name, attrs[attr_name].desc)
 
         self.desc.check_attrs()
-        self.desc.infer_shape(self.block.desc)
+        if type not in {'feed', 'fetch'}:
+            self.desc.infer_shape(self.block.desc)
```
Comment (tonyyang-svail) [inline python/paddle/v2/framework/framework.py]: why we can't infer shape a feed/fetch?

File: python/paddle/v2/framework/tests/test_executor_and_mul.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        tensor_b.set(b_np, place)
+        # del input_tensor
+        exe = Executor(place)
+        outs = list(
+            exe.run(g_program,
+                    feed={'a': tensor_a,
+                          'b': tensor_b},
+                    fetch_list=[out]))
+        out = numpy.array(outs[0])
+        self.assertEqual((100, 100), out.shape)
```
Comment (QiJune) [inline python/paddle/v2/framework/tests/test_executor_and_mul.py]: Better to check the result
```
self.assertTrue(numpy.allclose(out, numpy.dot(a_np, b_np)))
```

File: python/paddle/v2/framework/framework.py (hunk 1 above)
Comment (QiJune) [reply python/paddle/v2/framework/framework.py]: InferVarType is also needed.
```
self.desc.infer_var_type(self.block.desc)
```

File: paddle/operators/fetch_op.cc [hunk 3]
```diff
@@ -52,13 +52,25 @@ class FetchOp : public framework::OperatorBase {
...
 
+class FetchOpInfoMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  FetchOpInfoMaker(framework::OpProto *proto,
+                   framework::OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "The input of fetch op");
+    AddOutput("Out", "The output of fetch op");
+    AddComment("fetch op, it should not be configured by users directly");
+    AddAttr<int>("col", "column of feed");
```
Comment (QiJune) [inline paddle/operators/fetch_op.cc]: column of feed --> column of fetch

File: python/paddle/v2/framework/framework.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/framework/framework.py]: here `infer_shape` is a compile-time method. Feed/Fetch op is a run-time operator, do not need to infer shape.

Comment (reyoung) [reply python/paddle/v2/framework/framework.py]: > InferVarType is also needed.

Add in rest PRs

File: paddle/operators/fetch_op.cc (hunk 3 above)
Comment (reyoung) [reply paddle/operators/fetch_op.cc]: Done.

File: python/paddle/v2/framework/tests/test_executor_and_mul.py (hunk 2 above)
Comment (reyoung) [reply python/paddle/v2/framework/tests/test_executor_and_mul.py]: Done.

File: python/paddle/v2/framework/tests/test_executor_and_mul.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            append_batch_size=False)
+        out = mul(x=a, y=b)
+        place = core.CPUPlace()
+        a_np = numpy.random.random((100, 784)).astype('float32')
+        tensor_a = core.LoDTensor()
+        tensor_a.set(a_np, place)
+        b_np = numpy.random.random((784, 100)).astype('float32')
+        tensor_b = core.LoDTensor()
+        tensor_b.set(b_np, place)
+        # del input_tensor
```
[the remark this answers is not in the document]
Comment (reyoung) [reply python/paddle/v2/framework/tests/test_executor_and_mul.py]: Typo

File: python/paddle/v2/framework/executor.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        self.executor = core.Executor(act_places)
+
+    def run(self,
+            program,
+            feed,
+            fetch_list,
+            feed_var_name='feed',
+            fetch_var_name='fetch'):
+        if not isinstance(program, Program):
+            raise TypeError()
```
Comment (Canpio) [inline python/paddle/v2/framework/executor.py]: Error message?

**Pull request merged**